### DISCUSSION TITLE
Nemo feature request: de-emphasised text for hidden files, and/or add button to toolbar to toggle hidden files

### DISCUSSION THREAD

**Discussion by yushiyangk** [linuxmint/discussions#259] [Ideas]
It would be great if we could show hidden files, but keep it de-emphasised so that non-hidden files take more visual weight.

This would be similar to how Window's Explorer and macOS's Finder displays hidden files.

As a workaround, I could simply enable and disable showing hidden files on demand, but right now this involves several clicks through the menu. It would be great if I could add a single button to the toolbar to do this instantly as well.

Edit: I really do not understand why some of you get so *angry* about this right off the bat. Nobody's suggesting that this be a mandatory change; I definitely think it's a good idea to make it optional.

**Comment by rcalixte** [comment]
> It would be great if we could show hidden files, but keep it de-emphasised so that non-hidden files take more visual weight.

This is an interesting idea but hiding files in a Linux filesystem typically involves just having a period (.) as the first character, unlike Windows and macOS where it is an attribute of a file or directory. In that way, the visual cue for whether a file is hidden or not is serviced by identifying the first character. Though having another visual indicator as an option isn't a terrible idea.



> As a workaround, I could simply enable and disable showing hidden files on demand, but right now this involves several clicks through the menu. It would be great if I could add a single button to the toolbar to do this instantly as well.

A button might be a good addition but at the moment, there are two alternative options available that should hopefully be faster for you:
1. Using the mouse, you can right-click in a background area in Nemo and select 'Show Hidden Files':
![2023-12-20_22-22-51](https://github.com/linuxmint/discussions/assets/10281587/810c6cee-7212-43ab-a46d-55661b4c8e74)
(If this menu option is missing for you, you can make sure it is enabled in the menu by going to Edit -> Preferences -> Context Menus -> Background and then making sure that 'Show Hidden Files' is checked.)
![2023-12-20_22-27-01](https://github.com/linuxmint/discussions/assets/10281587/5965897f-fde3-4880-9c95-1ce87ba789f8)

2. Using the keyboard, you can press Ctrl+H to toggle whether hidden files are being shown. This is a default setting.

**Reply by yushiyangk** [reply]
Hi, I am aware that hidden files start with a dot. The point of this is not to provide an indicator for which files are hidden.

From a UX perspective, it is *much* faster to distinguish greyed out items at a glance.

This would allow the user to quickly identify which are "normal" files that they want to interact with while disregarding the greyed out files. At the same time, if the user specifically is interested in editing a hidden config file, he can do so simply by looking a bit closer, rather than having to go into the menu and temporarily enabling the display of hidden files.

With this enhancement, both flows can be supported effectively without requiring the user to toggle between the two states.

**Reply by rcalixte** [reply]
> From a UX perspective, it is _much_ faster to distinguish greyed out items at a glance.

Just a note that this may not be true for visually impaired or color blind users.

**Reply by yushiyangk** [reply]
I would argue even more so for visually impaired users, in the sense that the greyed out items are *less visible*, which is what we want.

Of course I'm not arguing that it should be invisible to visually impaired users. Making it optional would be a good idea.

**Comment by Secret-chest** [comment]
Should be optional!

**Comment by Crispycjr** [comment]
A toolbar toggle would be nice to: 

- Be able to visually discern whether hidden files are currently visible within a given directory
- Serve as a more convenient way to activate an essential file management feature for newer users, or just mouse-centric laptop users

The idea that this feature should be ignored because of the pre-existence of a keyboard shortcut seems fallacious, since just about all of the toolbar buttons with arguably much less general usage have shortcuts also attached to them, and the :eye_speech_bubble: icon is already there to serve as a good visual indicator for this button.

The 'grayed-out' hidden folders is honestly a huge feature, one of my few main gripes with Nemo. It's just much more visually functional and logical whether you're visually impaired or not, and especially harder to adjust to after getting used to Plasma's Dolphin.

**Comment by 100savage** [comment]
To make dot files slightly transparent, you can use a compositor like Compton or Picom.
```
sudo apt install compton
```
You'll need to configure the compositor to enable transparency for hidden files. This usually involves editing a configuration file. You can create it just for you at ~/.config/compton.conf or system wide at /etc/xdg/compton.conf
Here is an example of a compton.conf file with 25% transparency 

```
# Backend to use: "xrender" or "glx".
backend = "glx";

# Enable/disable shadows.
shadow = true;

# Opacity of window titlebars and borders.
# 0 means completely transparent, 1 means opaque.
# This applies to the entire window.
opacity-rule = [
  "0.75:class_g = 'URxvt'", # Example for terminal
  "0.75:class_g = 'Firefox'", # Example for Firefox
  "0.75:class_g = 'your-hidden-file-class'", # Example for hidden files
];

# Transparent windows, e.g., your hidden files will fall under this rule.
opacity-rule = [
  "0.25:class_g = 'hidden-file'"
];

# Enable fading of windows when opening/closing.
fading = true;

# How much fade to apply. Default is 0.8.
fade-delta = 4;

# The time in milliseconds between steps in a fade, 10 is default.
fade-in-step = 0.03;
fade-out-step = 0.03;

# Enabled with true, disabled with false.
mark-wmwin-focused = true;
mark-ovredir-focused = true;

# Provide shadows.
shadow = true;
no-dnd-shadow = true;
no-dock-shadow = true;

# Specify a list of conditions of windows that should have no shadow.
shadow-exclude = [
  "name = 'Notification'",
  "class_g = 'Conky'",
  "class_g = 'XScreenSaver'"
];

# Specify fade effect conditions.
fade-exclude = [ ];
```

**Reply by mtwebster** [reply]
What does this have to do with nemo? Did ChatGPT tell you this?

**Reply by 100savage** [reply]
I did not use ChatGPT, I did use Google and did a searched on how to make dot files transparent. I found several websites that talked about compton. Here are some of these websites.
 https://www.addictivetips.com/ubuntu-linux-tips/window-compositing-lightweight-linux-desktops-compton/
 https://github.com/chjj/compton
https://www.maketecheasier.com/speed-up-linux-desktop-with-compton/

You are correct in saying that what I found doesn't have to do with Nemo. I was just trying to help someone with a problem. I didn't find a way to make this happen in Nemo, but I thought that this could be something that can help someone out. If doing something like this is wrong... well I'm sorry for that, but if doing something like this solves their issue. They have the solution. If you being the maintainer feel that what I did is wrong, then just delete my account.

**Comment by cplittlfield** [comment]
For myself, I devoutly hope this would be an optional feature.  When I want to see the hidden files, I want to _see_ them.  If I don't need to see them, I hide them.  Ctrl + H works well as a shortcut, if you are tired of excessive mouse clicks.   What I don't understand about this request is why, if someone doesn't want to be disitracted by hidden files, they would even want them displayed in the first place.

In any case, this is an issue only with the Home folder, since the system doesn't store hidden files anywhere else.  Most people's user files are stored in sub-folders, where hidden files can't possibly be a distraction, because there aren't any.  Cluttering up one's Home folder with user files is just asking for trouble, in any case.

**Reply by yushiyangk** [reply]
>  What I don't understand about this request is why, if someone doesn't want to be disitracted by hidden files, they would even want them displayed in the first place.

So that you know that they are there, but don't have to pay attention to them. The same reason why *information hierarchy* is important in visual design, rather than outright *hiding information*.

[I have previously given a more detailed explanation here.](https://github.com/orgs/linuxmint/discussions/259#discussioncomment-7914276)

**Reply by cplittlfield** [reply]
I understand your logic, merely question the utility.

There are occasions when those "hidden" files precisely _are_ my focus, so how would I then make them fully visible, and take the focus off the regular files in such cases?

(I like the current setup precisely because when I want to focus on the hidden files, I can make them completely visible, and they all have a visual signifier, to indicate they are the files I want to be looking at.  On other occasions, minimizing the clutter by hiding such files altogether is very helpful.  I know they're still there, but I don't necessarily need to see them all the time.)

Also, given that the system puts hidden files only in /home/$USER/ and nowhere else, is there real utility in programming this feature just for use in a single directory?  How many directories do you have that contain hidden files?  And I'd be curious to know why, since I can't seem to think of a useful reason for doing so.

**Reply by yushiyangk** [reply]
Aren't we just circling back to the same points? You might be perfectly happy manually toggling it every time, but I would prefer not to have to do that, and would prefer one view that supports two different usage flows effectively.

Again, if it were hidden, the user is not necessarily aware that it exists and that it can be toggled to be visible. Making it visible but less prominent fixes that.

**Reply by Secret-chest** [reply]
> Again, if it were hidden, the user is not necessarily aware that it exists and that it can be toggled to be visible. Making it visible but less prominent fixes that.

Maybe a better way would be to count hidden files in parentheses next to the count in the status bar.

**Reply by yushiyangk** [reply]
@cplittlfield not sure why you replied in a different thread, but I'd just like to point out that no one is asking for this to be a mandatory change.

**Comment by JosephMcc** [comment]
Not a fan of this at all. The "greying" out of an item typically denotes that it's insensitive and not able to be reacted with. Plus these are easy enough to hide.

**Reply by yushiyangk** [reply]
Well perhaps instead of thinking of it as "greying out", we could think of it as "lower contrast". That is to say, not insensitive or disabled, merely less prominent.

Also, it's interesting to bring up disabled UI elements, because we can ask a similar question about those: why do we still show disabled UI elements that the user cannot interact with? Why grey it out at all, instead of hiding it?

I think the answer is that it 1) indicates to the user that that function potentially exists, and 2) maintains consistency in the amount of buttons visible and their positions. Both of these apply just as well to hidden files as a rationale for not hiding them all the time.

**Reply by yushiyangk** [reply]
@JosephMcc On the difference between "greying out" and "lower contrast", I'm not just being funny about it.

The use of "lower contrast" text already exists in Nemo!

If you take a look at the application menus in Nemo in LM 22.1, the keyboard shortcuts are indicated on the right of each menu entry in a *slightly lower contrast text*, precisely because it is less critical information. This lower contrast text is also notably different from "greyed out" text; the latter is an even lighter shade of grey.

![image](https://github.com/user-attachments/assets/27701257-5ea5-4a89-bb47-17daafdec2be)

This lower contrast text is less prominent, but still perfectly readable, and clearly distinguished from "greyed out" UI elements.

**Reply by yushiyangk** [reply]
I've edited the title of this thread to reflect this. You're definitely right that "greying out" was not giving the right impression.

**Reply by Secret-chest** [reply]
They're a different UI element though. All the accelerators are grey.

**Reply by cplittlfield** [reply]
"Aren't we just circling back to the same points?"  Yes, we are.

I've been following these posts and I have to say that the keyboard shortcuts in the Nemo menus can be very difficult to make out, for those of us with older eyes.  This is why I really don't want the hidden files made less visible, or if the devs see value in such a feature, I ask them to provide a way to TURN IT OFF.

Those of us with older eyes have to continually fight this battle with young designers who assume that everyone's vision is as good as theirs.

Okay, that's my point, for the last time.  I'm out.

**Reply by mtwebster** [reply]
I did a quick test of this in nemo, and I don't think 'greying' out would work reliably - depending on what font you use, the effect can either be insufficiently 'light' or so light that it affects its readability.

We utilize this effect in the Favorites location in the sidebar - files that are marked favorite but have been deleted show up in that view with a lighter font.
To reproduce:
- Mark some files in your home folder (or anywhere) as 'favorite' from the right-click menu.
- Close nemo
- In a *terminal*, delete one of those favorited files
- Open nemo, navigate to the Favorites location
- Observe the deleted file is listed, but with a lighter weight font.

It's not great, but it's isolated to this virtual folder alone - I don't think it can be improved enough to be used elsewhere.

One thing I recall in my years using Windows, is they'd use different font colors in the file manager to show some states - blue files were compressed, green encrypted I think... Personally I feel this would work better to make hidden files different than playing with font weight (though more prominent as well).

**Reply by Secret-chest** [reply]
Though colours wouldn't work well for all themes. If using colour, it should be themeable.

**Reply by mtwebster** [reply]
Neither would text with different weights.

**Reply by Secret-chest** [reply]
Yes, I was just pointing out. I believe such a feature ought to be optional because sometimes you may want to see the hidden files well.

**Reply by yushiyangk** [reply]
> They're a different UI element though. All the accelerators are grey.

The point is that "grey text" does not necessarily mean "greyed out/disabled control".

**Reply by Jopp-gh** [reply]
EX Mac user here. MacOs uses hidden folders and files everywhere.
And on Mac, you press cmd+shift+period to make invisible visible.

I wouldn't want to see invisible all times to clutter the home folder.
A shortcut is functional and quick, if I understood you correctly. 

Graphical gags like a grey tint for invisible files/folders made visible is bad, because many applications use to "gray out" buttons to mark them as inactive. Colors aren't just aesthetics, but to use as semantics in design choices first and in aesthetic plays after. 

If you want a solution, create a folder containing links to all invisible folders you need, so you don't have to switch invisibles on / off

**Reply by SuperninjaXII** [reply]
In UI reducing contrast indictes it's importance level, making it grey indicates inactive. A changed in folder icon can also be a solution so is the change in font weight and font style

**Reply by yushiyangk** [reply]
@Jopp-gh the entire discussion above you addresses why "grey text" does not necessarily mean "greyed out/disabled control". Please do read it.

> If you want a solution, create a folder containing links to all invisible folders you need, so you don't have to switch invisibles on / off 

Are you actually just being absurd on purpose?

**Reply by Jopp-gh** [reply]
On Mint, invisible folders / files aren't that important, like on Mac or windows. Hard to grasp?

So, put your important directories of invisible folders as links in some folder. Still hard to understand?

Add something constructive, instead to attack other people's view.

**Reply by yushiyangk** [reply]
> On Mint, invisible folders / files aren't that important

Unsubstantiated opinion.

> So, put your important directories of invisible folders as links in some folder.

No it is not reasonable to add a chore of manually arranging your hidden folders just to fight against the UI. There is nothing difficult to understand about it, it's just straight up absurd to suggest.

**Comment by Secret-chest** [comment]
What about exposing a CSS class for themes?
